File: Day 141/AI/binary.c
#include <stdio.h>

int binarySearch(int array[], int n, int target) {
    int low = 0;
    int high = n - 1;
    
    while (low <= high) {
        int mid = low + (high - low) / 2;
        int value = array[mid];
        
        if (value == target) {
            return mid;
        } else if (value < target) {
            low = mid + 1;
        } else {
            high = mid - 1;
        }
    }
    
    return -1; //Element not found
}

int main() {
    int array[] = {1, 2, 3, 4, 5, 6, 7, 8, 9};
    int n = sizeof(array) / sizeof(array[0]);
    int target = 5;
    
    int result = binarySearch(array, n, target);
    if (result != -1) {
        printf("Element found at index %d\n", result);
    } else {
        printf("Element not found\n");
    }
    
    return 0;
}

File: Day 141/AI/binary.js
function binarySearch(array, target) {
    let low = 0;
    let high = array.length - 1;

    while(low <= high) {
        const middle = Math.floor((low + high) / 2);
        if(array[middle] === target) {
            return middle;
        } else if(array[middle] < target) {
            low = middle + 1;
        } else {
            high = mid - 1;
        }
    }
    return -1; //element not found :-(
}

const array = [1, 4, 7, 5, 8];
const target = 4;
const index = binarySearch(array, target);